terraform/backend/module/alb.tf

```diff
@@ -1,12 +1,41 @@
+data "aws_region" "current" {}
+data "aws_caller_identity" "current" {}
+
+resource "aws_s3_bucket" "log-bucket" {
+  bucket_prefix = "${var.env}-${var.app}-external-alb-logs"
+}
+
+data "template_file" "log-bucket-policy" {
+  template = "${file("${path.module}/log-bucket-policy.json.tpl")}"
+
+  vars = {
+    alb_account_id = "${lookup(var.aws_region_map, "${data.aws_region.current.name}")}"
+    s3_arn         = "${aws_s3_bucket.log-bucket.arn}"
+    account_id     = "${data.aws_caller_identity.current.account_id}"
+  }
+
+resource "aws_s3_bucket_policy" "attach-policy-to-log-bucket" {
+  bucket = "${aws_s3_bucket.log-bucket.id}"
+  policy = "${data.template_file.log-bucket-policy.rendered}"
+}
+
+
 # alb
 resource "aws_lb" "backend" {
+  depends_on = [
+    "aws_s3_bucket.log-bucket",
+    "aws_s3_bucket_policy.attach-policy-to-log-bucket"
+  ]  
   name                             = "${var.environment}-backend"
   internal                         = false
   load_balancer_type               = "application"
   subnets                          = data.aws_subnet_ids.public_subnets.ids
   security_groups                  = [aws_security_group.lb_sg.id]
   enable_deletion_protection       = false
   enable_cross_zone_load_balancing = false
+  access_logs {
```

Comment:
can you make this optional with a var like enable_s3_access_logs and add it to the configs?